### PR TITLE
Create models

### PR DESCRIPTION
**Pull request by jorgeabrahan**
✅ Create user model class
✅ Create comment model class
✅ Create post model class
✅ Create like model class
✅ Create one-to-many relationship between comments and posts
✅ Create one-to-many relationship between comments and users
✅ Fix one-to-many relationship between comments and posts and users
✅ Create one-to-many relationship between post and user
✅ Create one-to-many relationship between likes and posts
✅ Create one-to-many relationship between users and likes
✅ Fix single line syntax between relationships
✅ Create tables foreign keys as references without indexes
✅ Solve the server connection lost error
✅ Create indexes for all foreign keys
✅ Add class name in relationship between models
✅ Create method to get 3 most recent posts for a user
✅ Create method to update the posts counter of a user
✅ Create method to get the most recent comments of a post
✅ Create method to update comments counter for a given post
✅ Create method to update the likes counter for a given post
✅ Run rails db:migrate

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  has_many :comments, class_name: 'Comment'
+  has_many :likes, class_name: 'Like'
+
+  belongs_to :users
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_comments_counter
```
Comment (rustam-tolipov) [inline app/models/post.rb]: **You did a great job :)**

- [x] However, I have a suggestion for you to update! For using the update comment counter method we can say that it only belongs to Comment and you do not need to use it in Posts, right? As this comes in handy, why not give it a try to put it after `private` and then call if after_save: update_com... I kindly suggest you do that :)

HINT: [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save

Comment (rustam-tolipov) [inline app/models/post.rb]: - [x] Also, please put update_comments_counter to it's own model where comments.rb

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    update(comments_counter: comments.count)
+  end
+
+  def update_likes_counter
```
Comment (rustam-tolipov) [inline app/models/post.rb]: - [x] The same requirement for update_likes_counter, put after private in it's own like.rb model

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :comments, class_name: 'Comment'
+  has_many :posts, class_name: 'Post'
+  has_many :likes, class_name: 'Like'
+
+  def recent_posts
```
Comment (rustam-tolipov) [inline app/models/user.rb:6]: - [x] The last but the same requirement. Put after private in post.rb

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @jorgeabrahan,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 3 above)
Comment (rustam-tolipov) [reply app/models/user.rb:6]: - [ ] If these are in screenshots are links, they are all broken. Kindly fix them and make sure there are no broken links in the readme. Thank you :)


Screenshot 2023-05-04 at 21 43 24

**Review by Alfredbis29 — APPROVED** [review]
# STATUS: APPROVED   
 ## Hi @jorgeabrahan 
Your project is complete! There is nothing else to say other than... Great job clap Congratulations! :tada:
 
 TO HIGHLIGHT:
 
 * Your project is professional :white_check_mark:
 * Well structured files :white_check_mark:
  * Good Job :+1: :smile: 
 
 Now you can merge this branch and move on 1st_place_medal: HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :


 
 ## Optional suggestions

- [ ] N/A keep Doing great work :clap:  :clap: 

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
 
Cheers and Happy coding! :clap:  :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**